libpod/boltdb_state.go

```diff
@@ -78,6 +80,17 @@ func NewBoltState(path string, runtime *Runtime) (State, error) {
 
 	logrus.Debugf("Initializing boltdb state at %s", path)
 
+	// BoltDB is deprecated and, as of Podman 5.0, we no longer allow the
+	// creation of new Bolt states.
+	// If the DB does not already exist, error out.
+	// To continue testing in CI, allow creation iff an undocumented env
+	// var is set.
+	if os.Getenv("ALLOW_LEGACY_DB") != "true" {
+		if _, err := os.Stat(path); err != nil && errors.Is(err, fs.ErrNotExist) {
+			return nil, fmt.Errorf("the BoltDB backend has been deprecated, no new BoltDB databases can be created: %w", define.ErrInvalidArg)
+		}
```

Comment:
Suggestion: flip the conditions (test exist first, and then env). Abort or warn:
```go
    if ... Stat, ErrNotExist {
        if ($CI_DESIRED_DATABASE==boltdb) {
            warn("BoltDB is deprecated; proceeding due to $CI_DESIRED_DATABASE. This is your only warning.")
        } else {
            return nil, Error(...)
        }
    }
```

Reply:
Can't really drop a warning here, unfortunately... We init Libpod multiple times per run on rootless, so we'd be spamming folks with multiple messages per run. I don't think there's an easy way around that, unfortunately, we don't really share anything between the runs.

Maybe a debug log, since those are already overly-verbose.

Reply:
There's still the hidden `--db-backend` flag.  Could we just continue keeping that around?  The new env variable confuses me as it adds (yet) another way to configure/set the DB.